### ISSUE TITLE
Can't convert sub properties in a complex object

### ISSUE THREAD

**Issue by bbcyyb** [agileobjects/AgileMapper#115]
Runnable code as below,
https://dotnetfiddle.net/6pcaqB

- Cannot convert a object C as property whose owner B belongs to a IList collection.
Looks like
```C#
class A { public IList Blst {get;set;} }
class B { public C Cc {get;set;}}
class C {public int Id{get;set;}}
```
the mapped property Cc in The target object is null

**Comment by SteveWilkes** [comment]
Hi!

`B`'s `CC` property is of type `List<C>`, but `BDto`'s `CC` property is of type `CDto` - the mapper doesn't know how to map from a `C` collection to a single `CDto`.

I've [updated](https://dotnetfiddle.net/dInpuS) the fiddle to show the mapping plan for `A` to `ADto`.

Cheers,

Steve

```C#
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// Map A -> ADto
// Rule Set: CreateNew
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 

aToAdData =>
{
    try
    {
        var aDto = new ADto();
        aDto.Id = aToAdData.Source.Id;

        if (aToAdData.Source.BB != null)
        {
            aDto.BB =
            {
                try
                {
                    var bs = aToAdData.Source.BB;
                    IList<BDto> bDtos = (aDto.BB != null)
                        ? aDto.BB.IsReadOnly ? new List<BDto>(aDto.BB) : aDto.BB
                        : new List<BDto>(bs.Count);
                    var i = 0;
                    while (true)
                    {
                        if (i == bs.Count)
                        {
                            break;
                        }

                        var b = bs[i];
                        bDtos.Add(
                            {
                                try
                                {
                                    if (b == null)
                                    {
                                        return null;
                                    }

                                    var bDto = new BDto();
                                    bDto.Id = b.Id;
                                    // No data source for CC or any of its child members

                                    return bDto;
                                }
                                catch (Exception ex)
                                {
                                    throw MappingException.For(
                                        "CreateNew",
                                        "A.BB[i]",
                                        "ADto.BB[i]",
                                        ex);
                                }
                            });
                        ++i;
                    }

                    return bDtos;
                }
                catch (Exception ex)
                {
                    throw MappingException.For(
                        "CreateNew",
                        "A.BB",
                        "ADto.BB",
                        ex);
                }
            }
        }
        else
        {
            aDto.BB = new List<BDto>();
        }

        return aDto;
    }
    catch (Exception ex)
    {
        throw MappingException.For(
            "CreateNew",
            "A",
            "ADto",
            ex);
    }
}
```

**Comment by bbcyyb** [comment]
Hi Steve,

Sorry, It's a clerical error, I think type of Cc property under BTO is IList<CDTo>, like this,

///////////////////////////////////////
    public class BDto
    {
        public int Id { get; set; }
        public IList<CDto> CC { get; set; }
    }

    public class CDto
    {
        public int Id { get; set; }

        public DDto DD { get; set; }
    }
///////////////////////////////////////

Is there any way to convert this kind of collecion object?

The new fiddle,
https://dotnetfiddle.net/42Ruvq

Thx,
Kevin,

**Comment by SteveWilkes** [comment]
Hi,

It looks like the `A` -> `ADto` case is working once we update the types, but `A2.B.CC` does look like it's being ignored in the `A2` -> `A2Dto` case. Thanks for letting me know, I'll look into it.

Cheers,

Steve

**Comment by bbcyyb** [comment]
Thanks Steve, looking forward to your reply

**Comment by SteveWilkes** [comment]
This is fixed in the latest code - it turned out to be an issue with the `C` -> `CDto` mapping being present on `B` and `A2`.

Preview release to follow - thanks very much for the feedback! :)

**Comment by SteveWilkes** [comment]
This is fixed in v1.1-preview5, which is now available [via NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks again! :)

**Comment by SteveWilkes** [comment]
This is fixed in [v1.1](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks! :)

**Comment by bbcyyb** [comment]
Hi Steve,

Sorry to reply you to this email for a long time. I went on holiday for the Chinese Spring Festival. I ran this sample and our real source code using the latest V1.1, it works very well. Thanks for you help Steve, have a good day!

Kevin,
